src/lib.rs

```diff
@@ -144,7 +144,7 @@ fn compare_descriptors(a: &ImageSampler, b: &ImageSampler) -> bool {
     match (a, b) {
         (ImageSampler::Default, ImageSampler::Default) => true,
         (ImageSampler::Descriptor(descriptor_a), ImageSampler::Descriptor(descriptor_b)) => {
-            descriptor_a == descriptor_b
+            descriptor_a.label == descriptor_b.label
```

Comment:
Idk if this might cause issues. Labels are optional so if multiple descriptors don't provide a label (or provide the same label) this could cause false equality checks. 
Not sure what the solution should be, we could push for bevy to implement partial eq like wgpu has or we could try to manually implement it. 
Also depending on what relies on this function, maybe it isn't important enough to worry about or is always used for descriptors we control and know are unique labels?

Reply:
compared all of them